src/api/spec/controllers/person_controller_spec.rb

```diff
@@ -20,4 +24,64 @@
       end
     end
   end
+
+  describe 'PUT #put_userinfo' do
+    let(:xml) {
+      <<-XML_DATA
+        <userinfo>
+          <realname>test name</realname>
+          <email>test@test.de</email>
+        </userinfo>
+      XML_DATA
+    }
+
+    context 'when in LDAP mode' do
+      before do
+        stub_const('CONFIG', CONFIG.merge({ 'ldap_mode' => :on }))
+        request.env["RAW_POST_DATA"] = xml
+      end
+
+      context 'as an admin' do
+        before do
+          login admin_user
+
+          put :put_userinfo, params: { login: user.login, format: :xml }
+          user.reload
+        end
+
+        it 'sets an error code' do
+          expect(response.header['X-Opensuse-Errorcode']).to eq('change_userinfo_no_permission')
+        end
+
+        it 'does not change users real name' do
+          expect(user.realname).to eq(old_realname)
+        end
+
+        it 'does not change users email address' do
+          expect(user.email).to eq(old_email)
+        end
```

Comment:
Lines 52-62 are the same as 73-83. I would use a [shared example](https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples).